applications/StructuralMechanicsApplication/custom_elements/solid_elements/small_displacement_mixed_strain_displacement_element.cpp

```diff
@@ -0,0 +1,1278 @@
+
+    Vector RHSu(dim * n_nodes), RHSe(strain_size * n_nodes);
+    noalias(RHSu) = ZeroVector(dim * n_nodes);
+    noalias(RHSe) = ZeroVector(strain_size * n_nodes);
+
+    Matrix K(dim * n_nodes, dim * n_nodes), G(n_nodes * strain_size, dim * n_nodes),
+        M(n_nodes * strain_size, n_nodes * strain_size), Q(dim * n_nodes, n_nodes * strain_size);
+    noalias(K) = ZeroMatrix(dim * n_nodes, dim * n_nodes);
+    noalias(G) = ZeroMatrix(n_nodes * strain_size, dim * n_nodes);
+    noalias(Q) = ZeroMatrix(dim * n_nodes, n_nodes * strain_size);
+    noalias(M) = ZeroMatrix(n_nodes * strain_size, n_nodes * strain_size);
+
+    const auto& r_integration_points = GetGeometry().IntegrationPoints(mThisIntegrationMethod);
+    SizeType n_gauss = r_integration_points.size();
+
+    Matrix D0(strain_size, strain_size);
+    mConstitutiveLawVector[0]->CalculateValue(cons_law_values, CONSTITUTIVE_MATRIX, D0);
+
+    // Gauss IP loop
+    for (IndexType i_gauss = 0; i_gauss < n_gauss; ++i_gauss) {
+
+        const auto body_force = GetBodyForce(r_geometry.IntegrationPoints(GetIntegrationMethod()), i_gauss);
+
+        CalculateKinematicVariables(kinematic_variables, i_gauss, GetIntegrationMethod());
+
+        double w_gauss = kinematic_variables.detJ0 * r_integration_points[i_gauss].Weight();
+        if (dim == 2 && r_props.Has(THICKNESS))
+            w_gauss *= r_props[THICKNESS];
+
+        noalias(constitutive_variables.StrainVector) = kinematic_variables.EquivalentStrain;
+
+        // Calculate the constitutive response with the equivalent stabilized strain
+        CalculateConstitutiveVariables(kinematic_variables,  constitutive_variables, cons_law_values, i_gauss, r_integration_points, ConstitutiveLaw::StressMeasure_Cauchy);
+
+        // Now we add the body forces
+        for (IndexType i = 0; i < n_nodes; ++i) {
+            const SizeType index = dim * i;
+            for (IndexType j = 0; j < dim; ++j)
+                RHSu[index + j] += w_gauss * kinematic_variables.N[i] * body_force[j];
+        }
+
+        // Contributions to the RHS
+        noalias(RHSu) -= w_gauss * prod(trans(kinematic_variables.B), constitutive_variables.StressVector);
+        const Vector strain_diff = kinematic_variables.SymmGradientDispl - kinematic_variables.EquivalentStrain;
+        noalias(RHSe) -= w_gauss * prod(trans(kinematic_variables.N_epsilon), Vector(prod(D0, strain_diff)));
+
+        // Contributions to the LHS
+        noalias(K) += w_gauss * prod(trans(kinematic_variables.B), Matrix(prod(constitutive_variables.D, kinematic_variables.B)));
+        noalias(Q) += w_gauss * prod(trans(kinematic_variables.B), Matrix(prod(constitutive_variables.D, kinematic_variables.N_epsilon)));
+
+        noalias(M) += w_gauss * prod(trans(kinematic_variables.N_epsilon), Matrix(prod(D0, kinematic_variables.N_epsilon)));
+        noalias(G) += w_gauss * prod(trans(kinematic_variables.N_epsilon), Matrix(prod(D0, kinematic_variables.B)));
```

Comment:
I am not 100% sure this is the most efficient. Now with the benchmark it is possible to test it.